module-gui/gui/widgets/status-bar/SIM.cpp

```diff
@@ -31,6 +31,12 @@ namespace gui::status_bar
             [[fallthrough]];
         case GSM::SIM::NONE:
             [[fallthrough]];
+        case GSM::SIM::SIM_NEED_PIN:
```

Comment:
On the contrary, having the status bar here, I think this icon is perfectly legitimate: having no more space for details, we should just display as if the card wasn't connected, so that's pretty OK.